server/game/gameSystems/ViewCardSystem.ts

```diff
@@ -0,0 +1,76 @@
+import { AbilityContext } from '../core/ability/AbilityContext';
+import { BaseCard } from '../core/card/BaseCard';
+import { ViewCardType } from '../core/Constants';
+import { GameEvent } from '../core/event/GameEvent';
+import { CardTargetSystem, ICardTargetSystemProperties } from '../core/gameSystem/CardTargetSystem';
+import Player from '../core/Player';
+
+export interface IViewCardProperties extends ICardTargetSystemProperties {
+    viewType: ViewCardType;
+    sendChatMessage?: boolean;
```

Comment:
Is there any situation where we wouldn't send a chat message currently, or are you putting this here for when we add a UI for this later and the message is then optional?

Reply:
Actually I just thought of the answer to my own question - in the case when the opponent is looking at top card of a deck (e.g. Thrawn), that shouldn't be broadcast. For now we can leave this as-is since we have nothing in place for the UI but we should probably leave something like TODO LOOKAT here to remember

Reply:
@AMMayberry1 Yeah, it looks like all lookAt/reveal logic is going through the chat log atm, which is not going to work for Thrawn or for Twin Suns in the future if we support that. We need some sort of popup that shows info to only certain players